### PR TITLE
Cleanup rules.mk for 32U4 keyboards, A-F

### PR DESCRIPTION
**Pull request by fauxpark**
## Description

Here we go... part 1 of 4.

Some of the boards without existing `BOOTLOADER`s set I had to guess at, based on what Google gave me, but I think they're pretty good guesses :) it won't let the magic smoke out anyway.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by fauxpark** [comment]
I didn't change any of that, just moved those lines down below the build options.

**Comment by zvecr** [comment]
I guess at a 2nd look, 

`keyboards/aeboards/aegis/keymaps/via/rules.mk`
```diff
- # project specific files
- SRC =	keyboards/wilba_tech/wt_main.c
```
and it was relocated from top to bottom of `keyboards/aeboards/aegis/rules.mk`.

Overall not a behaviour change, as the file was always being compiled, though my understanding is  not ideal as it bleeds out of the via keymap.

**Comment by fauxpark** [comment]
Yeah, I didn't want to touch that stuff other than deduplicating it, but perhaps the wt_main includes should go only in the via keymaps?

File: keyboards/alu84/rules.mk [hunk 1]
```diff
@@ -53,4 +23,4 @@ SLEEP_LED_ENABLE ?= yes  # Breathing sleep LED during USB suspend
 NKRO_ENABLE ?= yes		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
 BACKLIGHT_ENABLE ?= yes  # Enable keyboard backlight functionality
 AUDIO_ENABLE ?= no
-RGBLIGHT_ENABLE ?= yes
\ No newline at end of file
+RGBLIGHT_ENABLE ?= yes
```
Comment (yanfali) [inline keyboards/alu84/rules.mk]: get rid of ?= maybe?

File: keyboards/butterstick/rules.mk [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
 MOUSEKEY_ENABLE 	= yes		# Mouse keys(+4700)
 EXTRAKEY_ENABLE 	= yes       # Audio control and System control(+450)
 CONSOLE_ENABLE 		= yes       # Console for debug(+400)
 COMMAND_ENABLE 		= no		# Commands for debug and configuration
 NKRO_ENABLE 		= yes       # USB Nkey Rollover
 STENO_ENABLE		= yes 		# Needed for chording
 
+OPT_DEFS += -DONLYQWERTY -DDEBUG_MATRIX
+SRC += sten.c
+EXTRAFLAGS += -flto
```
Comment (yanfali) [inline keyboards/butterstick/rules.mk]: convert to new flag maybe? LINK_TIME_OPTIMIZATION_ENABLE

File: keyboards/chimera_ergo/rules.mk [hunk 3]
```diff
@@ -78,4 +28,12 @@ NKRO_ENABLE = yes		# USB Nkey Rollover - not yet supported in LUFA
 UNICODE_ENABLE = YES 		# Unicode
 # BLUETOOTH_ENABLE = yes # Enable Bluetooth with the Adafruit EZ-Key HID
 
+
+OPT_DEFS += -DCHIMERA_ERGO_PROMICRO
+CHIMERA_ERGO_UPLOAD_COMMAND = while [ ! -r $(USB) ]; do sleep 1; done; \
```
Comment (yanfali) [inline keyboards/chimera_ergo/rules.mk]: is this required anymore?

File: keyboards/clueboard/66_hotswap/prototype/rules.mk [hunk 4]
```diff
@@ -22,3 +26,6 @@ RGBLIGHT_ENABLE = yes
 MIDI_ENABLE = no
 UNICODE_ENABLE = no
 BLUETOOTH_ENABLE = no
+
+EXTRAFLAGS += -flto
```
Comment (yanfali) [inline keyboards/clueboard/66_hotswap/prototype/rules.mk]: LINK_TIME_OPTIMIZATION_ENABLE

File: keyboards/crkbd/rules.mk [hunk 5]
```diff
@@ -70,5 +32,11 @@ RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight.
 SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
 
 CUSTOM_MATRIX = yes
+QUANTUM_LIB_SRC += 	i2c.c \
+					serial.c
```
Comment (yanfali) [inline keyboards/crkbd/rules.mk]: let's fix spacing

File: keyboards/ergodone/rules.mk [hunk 6]
```diff
@@ -92,4 +36,10 @@ SWAP_HANDS_ENABLE       = no  # Disable Onehand
 RGBLIGHT_ENABLE         = no
 MIDI_ENABLE             = no
 
+# project specific files
+SRC = \
+    twimaster.c \
```
Comment (yanfali) [inline keyboards/ergodone/rules.mk]: spacing

File: keyboards/alu84/rules.mk (hunk 1 above)
Comment (noroadsleft) [reply keyboards/alu84/rules.mk]: This whole section is using `?=`. That should be changed IMO.

File: keyboards/ai03/orbit/rules.mk [hunk 7]
```diff
@@ -88,4 +35,8 @@ USE_I2C = no                # I2C for split communication
 CUSTOM_MATRIX = yes			# For providing custom matrix.c (in this case, override regular matrix.c with split matrix.c)
 # SPLIT_KEYBOARD = yes		# Split keyboard flag disabled as manual edits had to be done to the split common files
 
-
+SRC += split_util.c \
```
Comment (noroadsleft) [inline keyboards/ai03/orbit/rules.mk]: This section is mixing tabs and spaces.

File: keyboards/alu84/rules.mk (hunk 1 above)
Comment (fauxpark) [reply keyboards/alu84/rules.mk]: I didn't want to mess with the build options sections here, so I left that stuff alone, as much as it bothers me 😄

**Comment by drashna** [comment]
Maybe push this to breaking changes then?  Give time to refactor if needed.

**Comment by fauxpark** [comment]
I can revert the `via` keymaps for now... but I'm interested to know those "reasons". These settings are mostly keyboard-specific things, except of course for handwires with multiple compatible controllers eg. PM/Elite/Proton-C. So I don't know what the goal is there.

**Comment by fauxpark** [comment]
Undid the VIA keymaps. Should be good to go now, I think?

File: keyboards/comet46/rules.mk [hunk 8]
```diff
@@ -64,3 +29,7 @@ UNICODE_ENABLE = YES 		# Unicode
 # BLUETOOTH_ENABLE = yes # Enable Bluetooth with the Adafruit EZ-Key HID
 
 USB = /dev/ttyACM0
+# project specific files
+SRC += matrix.c \
+       i2c.c \
```
Comment (drashna) [inline keyboards/comet46/rules.mk]: i2c.c should be added via `QUANTUM_SRC_LIB`, to avoid issues.  

Howver, that may be outside the scope of this PR.

**Pull request merged**